shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMQOperation.java

```diff
@@ -172,11 +172,11 @@ public void handleDelivery(String consumerTag,
                            log.debug(String.format("Message received : %s. delivery Tag : %s.", message, deliveryTag));
                         } catch (Exception e) {
                             log.error("Node-message processing failed.", e);
-                            // TODO : NACK
+                            channel.basicNack(deliveryTag,false,true);
```

Comment:
Did you tested this NAck ? Is it pushing back to queue ? Also, we need to check feature like visibility timeout in rabbit. Reference : https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html

If there is no such feature in Rabbit, then it may harm Shepherd threads. We need to come with some approach for the same.